scripts/ci/pre_commit/pre_commit_base_operator_partial_arguments.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import ast
+import itertools
+import pathlib
+import sys
+import typing
+
+ROOT_DIR = pathlib.Path(__file__, "..", "..", "..", "..").resolve()
```

Comment:
NIT:
```suggestion
ROOT_DIR = pathlib.Path(__file__).parents[3].resolve()
```

Reply:
FWIW, this is arguably slightly faster than `parents` because it does not need to build the extra objects. Also the direct equivalent should be `.resolve().parents[3]` since `__file__` may not contain the full parent chain.